cmd/bosun/sched/check.go

```diff
@@ -185,9 +201,87 @@ func (s *Schedule) RunHistory(r *RunHistory) {
 	if checkNotify && s.nc != nil {
 		s.nc <- true
 	}
+	s.CollectStates()
 	s.Save()
 }
 
+// CollectStates sends various state information to bosun with collect.
+func (s *Schedule) CollectStates() {
+	// [AlertName][Severity]Count
+	severityCounts := make(map[string]map[string]int64)
+	abnormalCounts := make(map[string]map[string]int64)
+	ackStatusCounts := make(map[string]map[bool]int64)
+	activeStatusCounts := make(map[string]map[bool]int64)
+	// Initalize the Counts
+	for _, alert := range s.Conf.Alerts {
+		severityCounts[alert.Name] = make(map[string]int64)
+		abnormalCounts[alert.Name] = make(map[string]int64)
+		for _, severity := range StatusNames {
```

Comment:
I think it's bad to do this this way. I would instead recommend making a statusLast status, and then loop from 0 to statusLast and convert it to a string. This way you don't need that silly slice.